processor/src/test/java/org/mapstruct/ap/testutil/WithServiceImplementation.java

```diff
@@ -0,0 +1,44 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.testutil;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Specifies an implementation of an SPI to be used during the annotation processing.
+ *
+ * @author Andreas Gudian
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface WithServiceImplementation {
+    /**
+     * @return The service implementation class that is to be made available during the annotation processing.
+     */
+    Class<?> value();
+
+    /**
+     * @return The SPI that the service implementation provides. If omitted, then {@link #value()} is expected to
+     *         implement exactly one interface which will be considered the provided SPI implementation.
+     */
+    Class<?> provides() default Object.class;
```

Comment:
now.. this is what I like.. :smile: . smartly designed to take scan for an interface on the value..